<launch>

  <arg name="bag_path" default="" /> 
  <arg name="save_dir" default="" />
  <arg name="publish_framerate" default="60.0" />
  
  <!-- pure event reconstruction -->
  <node name="reconstruction_node" pkg="pure_event_reconstruction" type="reconstruction_node" output="screen" args="--v=3">
    <param name="working_dir" value="$(env PWD)"/>
    <param name="bag_path" type="str" value="$(arg bag_path)"/>
    <param name="save_dir" type="str" value="$(arg save_dir)"/>
	<param name="publish_framerate" type="double" value="$(arg publish_framerate)"/>
	
    <remap from="events" to="dvs/events" />
  </node>
  
  <!-- display -->
  <node name="image_view" pkg="image_view" type="image_view">
    <remap from="image" to="pure_event_reconstruction/intensity_estimate"/>
  </node>

  <!-- configure -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" if="$(eval arg('bag_path') == '')"/> 

</launch>
